### PR TITLE
[DX] Move diffs right to the code

### PR DESCRIPTION
**Pull request by TomasVotruba**
## Before

The longer the content, the more scrolling to find the diff :face_exhaling: 

Screenshot from 2024-08-02 10-25-22

---

## After

Diff is right in the code, so no more scrolling.

Screenshot from 2024-07-15 01-31-03

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Note: add light + dark theme of diffs .)

File: src/Entity/AbstractRectorRun.php [hunk 1]
```diff
@@ -27,9 +29,33 @@ public function getUuid(): Uuid
...
+    /**
+     * @return DiffSnippet[]
+     */
+    public function getDiffSnippets(): array
+    {
+        $diffSnippets = [];
+
+        $contentDiffsRaw = Strings::split($this->getContentDiff(), '#@@ #');
+        foreach (array_filter($contentDiffsRaw) as $contentDiffRaw) {
+            [$lines, $content] = Strings::split($contentDiffRaw, "# @@\n#");
```
Comment (samsonasik) [inline src/Entity/AbstractRectorRun.php:41]: it seems cause error:

```
 Undefined array key 1 
```

Screenshot 2024-08-02 at 18 45 25

with code example:

```php
<?php

namespace Miofair\LayoutBundle\Tests\Form\Type;

use PHPUnit\Framework\MockObject\MockObject;
use PHPUnit\Framework\TestCase;

class OnCallbackTest extends TestCase
{
    private (\Exception&MockObject)|null $mock = null;

    protected function setUp(): void
    {
        $this->mock = $this->createMock(\Exception::class);
    }

    public function testConfigureOptions(): void
    {
        $matcher = static::exactly(6);
        $this->mock
            ->expects($matcher)
            ->method('__toString')
            ->willReturnCallback(function (){
                return $this->mock;
            })
        ;
    }
}
```

and code config:

```php
<?php

use Rector\Config\RectorConfig;
use Rector\TypeDeclaration\Rector\Property\TypedPropertyFromAssignsRector;

return RectorConfig::configure()
    // A. whole set
 //   ->withPreparedSets(typeDeclarations: true)
    // B. or few rules
    ->withRules([
        TypedPropertyFromAssignsRector::class
    ]);
```

Comment (samsonasik) [reply src/Entity/AbstractRectorRun.php:41]: It happen when content result is:

```
"// no change"
```

Comment (samsonasik) [reply src/Entity/AbstractRectorRun.php:41]: https://github.com/rectorphp/getrector-com/pull/2488

**Pull request merged**